common/src/main/java/com/wynntils/features/user/GammabrightFeature.java

```diff
@@ -23,65 +20,20 @@ public class GammabrightFeature extends UserFeature {
     @Config
     private boolean gammabrightEnabled = false;
 
-    @Config(visible = false)
-    private double lastGamma = 1f;
-
     @RegisterKeyBind
     private final KeyBind gammabrightKeyBind =
             new KeyBind("Gammabright", GLFW.GLFW_KEY_G, true, this::toggleGammaBright);
 
     @SubscribeEvent
-    public void onWorldStateChange(WorldStateEvent event) {
-        if (event.getNewState() != WorldStateManager.State.WORLD) return;
-
-        applyGammabright();
-    }
-
-    @Override
-    protected void onConfigUpdate(ConfigHolder configHolder) {
-        if (configHolder.getFieldName().equals("gammabrightEnabled")) {
-            applyGammabright();
-        }
-    }
-
-    @Override
-    protected void onDisable() {
-        resetGamma();
-    }
-
-    @Override
-    protected boolean onEnable() {
-        if (gammabrightEnabled && McUtils.options().gamma != 1000d) {
-            enableGammabright();
-        }
+    public void onUpdateLightTexture(UpdateLightTextureEvent event) {
+        if (!gammabrightEnabled) return;
 
-        return true;
-    }
-
-    private void applyGammabright() {
-        if (!isEnabled()) return;
-        if (gammabrightEnabled && McUtils.options().gamma == 1000d) return;
-
-        if (gammabrightEnabled) {
-            enableGammabright();
-        } else {
-            resetGamma();
-        }
+        event.setGamma(1000f);
     }
 
     private void toggleGammaBright() {
         gammabrightEnabled = !gammabrightEnabled;
-        applyGammabright();
 
         Managers.Config.saveConfig();
     }
-
-    private void resetGamma() {
-        McUtils.options().gamma = lastGamma;
-    }
-
-    private void enableGammabright() {
-        lastGamma = McUtils.options().gamma;
-        McUtils.options().gamma = 1000d;
```

Comment:
So there is no longer a gamma variable, but the gamma value is returned set when you convert a Double to float in `updateLightTexture`? It seems a bit... I dunno. Maybe this is the only way, but I don't like `@Redirect`. This sounds like it will break with other mods that change gamma, like Tweakeroo.

Reply:
*reading the code* Oh, I see, they had to rewrite this to be more dynamic because of the Warden.

Hm.

Reply:
If I look at the correct sources, there is still a:
```
                        float v = ((Double)this.minecraft.options.gamma().get()).floatValue();
```

Is that what you are trying to target with the mixin? But I think you will get this instead, that occurs before. Or is the mixin injected in both places in such a case?
```
                float h = ((Double)this.minecraft.options.darknessEffectScale().get()).floatValue();
```

Reply:
I think it would be safer and more robust to hijack
```
    public OptionInstance<Double> gamma() {
        return this.gamma;
    }
```
in Options instead. Maybe send a GammaRequestedEvent with the option to supply a replacement OptionInstance?

Reply:
Or, maybe we could just do `minecraft.options.gamma().set(100.0f)`? I think that ought to work. That would probably be a straightforward port of the old code. Save the old gamma value by `minecraft.options.gamma().get()` and then update it using the setter.

Reply:
> Or, maybe we could just do `minecraft.options.gamma().set(100.0f)`? I think that ought to work. That would probably be a straightforward port of the old code. Save the old gamma value by `minecraft.options.gamma().get()` and then update it using the setter.

No, config options for doubles are limited between 0-1.

Reply:
Hacking options is a way, I tried it. It is worse than this. I can agree with perhaps hijacking the config option itself. Although, that is hard because of the OptionsInstance wrapper. This might be the best we can do.

Reply:
MC would complain on config load

Reply:
And no, I don't agree that the situation where we send out an event each time we try to read the event variable is simpler, than updating the value when requested. It might be that this is the only reasonable way forward, but I still think the old way was better and I'd like to not let that go yet.

I will look into Tweakeroo to see how they do gamma bright on 1.19...

Reply:
Fine. I guess we need to revert this and I'll do the mixin.

Reply:
Thanks!